lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,105 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.raw_current=(raw_current)
+          Thread.current[:attr_raw_current] = raw_current
+        end
+
+        def self.raw_current
+          Thread.current[:attr_raw_current]
+        end
+
+        def self.current
+          raise "Current User has not been set" unless Thread.current[:attr_current_user]
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user)
+          raise StandardError, "Must be a ManageIQ::API::Common::User" unless user.is_a?(ManageIQ::API::Common::User)
+          Thread.current[:attr_current_user] = user
+        end
+
+        def self.from_header(raw_header)
+          self.raw_current = raw_header
+          raw_header.nil? ? Thread.current[:attr_current_user] = nil : self.current = new(raw_current)
+        end
+
+        def self.with_logged_in_user(user_header)
+          self.from_header(user_header)
+          yield self.current
+        ensure
+          Thread.current[:attr_current_user] = nil
+        end
+
+        def initialize(user_header)
+          raise StandardError, "Must pass in a valid Header or User Hash"  unless user_header.present?
+          @user = user_header
+        end
+
+        def username
+          validate_and_return(__method__)
+        end
+
+        def email
+          validate_and_return(__method__)
+        end
+
+        def first_name
+          validate_and_return(__method__)
+        end
+
+        def last_name
+          validate_and_return(__method__)
+        end
+
+        def is_active?
+          validate_and_return(:is_active)
+        end
+
+        def is_org_admin?
+          validate_and_return(:is_org_admin)
+        end
+
+        def is_internal?
+          validate_and_return(:is_internal)
+        end
+
+        def locale
+          validate_and_return(__method__)
+        end
+
+        def tenant
+          result = decode.dig('identity','account_number')
+          raise StandardError "Tenant key doesn't exist" if result.nil?
```

Comment:
No idea how this is working, since the comma is missing...I would have expected a SyntaxError.

Reply:
Dug into this, and it's a total coincidence this isn't a SyntaxError.  This code, if you try to execute it, gives...

```ruby
irb(main):001:0> def aaa
irb(main):002:1> raise StandardError "xxx"
irb(main):003:1> end
irb(main):004:0> aaa
NoMethodError: undefined method `StandardError' for main:Object
    from (irb):2:in `aaa'
    from (irb):4
    from /home/bdunne/.rubies/ruby-2.4.5/bin/irb:11:in `<main>'
```

I think specs will catch this.  Does this also explain the NoMethodError question I had elsewhere?